src/codegen/struct-or-class.lisp

```diff
@@ -43,43 +43,48 @@
         (list
          `(defstruct (,classname
                       (:copier nil)
+                      (:predicate nil)
                       ,@(when superclass
                           (list `(:include ,superclass)))
-                      (:constructor ,constructor ,field-names)
-                      (:predicate nil))
+                      (:constructor ,constructor ,field-names)) 
             ,@(loop :for field :in fields
                     :for name := (struct-or-class-field-name field)
                     :collect `(,name (error ""))))))
 
        (:class
-        (list 
-         `(defclass ,classname
-              ,(if superclass
-                   (list superclass)
-                   (list))
-            ,(loop :for field :in fields
-                   :for name := (struct-or-class-field-name field)
-                   :for package := (symbol-package classname)
-                   :for accessor
-                     := (alexandria:format-symbol package "~A-~A" classname name)
-                   :collect `(,name
-                              :initform (error "")
-                              :initarg ,name
-                              :accessor ,accessor)))
+        (append 
+         (list
+          `(defclass ,classname
+               ,(if superclass
+                    (list superclass)
+                    (list))
+             ,(loop :for field :in fields
+                    :for name := (struct-or-class-field-name field)
+                    :for package := (symbol-package classname)
+                    :for accessor
+                      := (alexandria:format-symbol package "~A-~A" classname name)
+                    :collect `(,name
+                               :initform (error "")
```

Comment:
I would prefer `:default-initarg` over `:initform`, also with an actual error string (or maybe some sort of `(unreachable)` call)

Reply:
why do you prefer `:default-initarg` over `:initform`?

Reply:
habit from [this tip](https://lisptips.com/post/11728375873/initform-and-default-initargs/amp)